actionpack/lib/action_dispatch/journey/route.rb

```diff
@@ -10,12 +10,12 @@ class Route
       module VerbMatchers
         VERBS = %w{ DELETE GET HEAD OPTIONS LINK PATCH POST PUT TRACE UNLINK }
         VERBS.each do |v|
-          class_eval <<-eoc
-          class #{v}
-            def self.verb; name.split("::").last; end
-            def self.call(req); req.#{v.downcase}?; end
-          end
-          eoc
+          class_eval <<-EOC, __FILE__, __LINE__ + 1
```

Comment:
This helps to know source location of a method

(before)
```ruby
> ActionDispatch::Journey::Route::VerbMatchers::PUT.method(:verb).source_location
["(eval)", 2]
```
(after)
```ruby
> ActionDispatch::Journey::Route::VerbMatchers::PUT.method(:verb).source_location
=> ["/path/../lib/action_dispatch/journey/route.rb", 15]
```
We always provide params filename and lineno as \_\_FILE\_\_, \_\_LINE\_\_ + 1 in accordance if class_eval with <<.

I believe example above is one of the reasons to add these params and why we pass them everywhere